test/SlowTests/Issues/RavenDB_19548.cs

```diff
@@ -0,0 +1,100 @@
+                }
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<TestObj>() select result.Prop1.Length;
+                    Assert.Equal(6, await asyncDocumentQuery.SingleAsync());
+                    Assert.Equal("from 'TestObjs' select Prop1.Length"
+                        , asyncDocumentQuery.ToString());
+                }
+
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<TestObj>()
+                                             let ret = RavenQuery.Raw<int>("result.Prop1.length")
+                                             select ret;
+
+                    var i = await asyncDocumentQuery.SingleAsync();
+                    Assert.Equal(6, i);
+                    Assert.Equal("declare function output(result) {\r\n\tvar ret = result.Prop1.length;\r\n\treturn { ret : ret };\r\n}\r\nfrom 'TestObjs' as result select output(result)"
+                        , asyncDocumentQuery.ToString());
+
+                }
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<TestObj>()
+                                             let ret = RavenQuery.Raw<int>("result.Prop1.length")
+                                             let ret2 = RavenQuery.Raw<int>("result.Prop2.length")
+                                             let x = ret + ret2
+                                             select x;
+
+                    var i = await asyncDocumentQuery.SingleAsync();
+                    Assert.Equal(8, i);
+                    Assert.Equal("declare function output(result) {\r\n\tvar ret = result.Prop1.length;\r\n\tvar ret2 = result.Prop2.length;\r\n\tvar x = ret+ret2;\r\n\treturn { x : x };\r\n}\r\nfrom 'TestObjs' as result select output(result)"
+                        , asyncDocumentQuery.ToString());
+                }
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<TestObj>()
+                                             let ret = RavenQuery.Raw<string>("result.Name.substr(0,3)")
+                                             select ret;
+
+                    var queryResult = await asyncDocumentQuery.ToListAsync();
+                    Assert.Equal("Ome", queryResult[0]);
+                    Assert.Equal("declare function output(result) {\r\n\tvar ret = result.Name.substr(0,3);\r\n\treturn { ret : ret };\r\n}\r\nfrom 'TestObjs' as result select output(result)"
+                        , asyncDocumentQuery.ToString());
+                }
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<TestObj>()
```

Comment:
do we want to add support for the following query?

```
var asyncDocumentQuery = from result in session.Query<TestObj>()
                select RavenQuery.Raw<int>("result.Prop1.length");
```

Reply:
Done